primitives/core/src/crypto.rs

```diff
@@ -354,16 +359,16 @@ pub trait Derive: Sized {
 	/// Derive a child key from a series of given junctions.
 	///
 	/// Will be `None` for public keys if there are any hard junctions in there.
-	#[cfg(feature = "std")]
+	#[cfg(all(feature = "serde", feature = "full_crypto"))]
 	fn derive<Iter: Iterator<Item = DeriveJunction>>(&self, _path: Iter) -> Option<Self> {
 		None
 	}
 }
 
-#[cfg(feature = "std")]
+#[cfg(all(feature = "serde", feature = "full_crypto"))]
 const PREFIX: &[u8] = b"SS58PRE";
 
-#[cfg(feature = "std")]
+#[cfg(all(feature = "serde", feature = "full_crypto"))]
 fn ss58hash(data: &[u8]) -> Vec<u8> {
 	use blake2::{Blake2b512, Digest};
```

Comment:
All of this is really not related to `serde`. I think you can revert most of the `cfg` changes here.

Reply:
`from_ss58check()` needed by ecdsa `Public` deserialization needs the function `ss58hash`